CodePush.m

```diff
@@ -36,26 +36,41 @@ @implementation CodePush {
 // that is associated with an update's package.
 static NSString *const PackageHashKey = @"packageHash";
 static NSString *const PackageIsPendingKey = @"isPending";
+static NSString *const BinaryBundleDateKey = @"binaryDate";
+
+// These values are used to save the bundleURL and extension for the JS bundle
+// in the binary.
+static NSString *binaryJsName = @"main";
+static NSString *binaryJsExtension = @"jsbundle";
+
 
 #pragma mark - Public Obj-C API
 
++ (NSURL *)binaryJsBundleUrl
+{
+    return [[NSBundle mainBundle] URLForResource:binaryJsName withExtension:binaryJsExtension];
+}
+
 + (NSURL *)bundleURL
 {
-    return [self bundleURLForResource:@"main"];
+    return [self bundleURLForResource:binaryJsName];
 }
 
 + (NSURL *)bundleURLForResource:(NSString *)resourceName
 {
+    binaryJsName = resourceName;
     return [self bundleURLForResource:resourceName
-                        withExtension:@"jsbundle"];
+                        withExtension:binaryJsExtension];
 }
 
 + (NSURL *)bundleURLForResource:(NSString *)resourceName
                   withExtension:(NSString *)resourceExtension
 {
+    binaryJsName = resourceName;
+    binaryJsExtension = resourceExtension;
     NSError *error;
     NSString *packageFile = [CodePushPackage getCurrentPackageBundlePath:&error];
-    NSURL *binaryJsBundleUrl = [[NSBundle mainBundle] URLForResource:resourceName withExtension:resourceExtension];
+    NSURL *binaryJsBundleUrl = [self binaryJsBundleUrl];
```

Comment:
It feels a little weird to set these two variables and then use the `binaryJsBundleUrl` to then access them. I like the previous line better since it's clearer what is happening.